### PR TITLE
Improve client API

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Depends on #554

* Convert getters into properties for `CustomSerializable`, `Serializer`, `Portable`, `IdentifiedDataSerializable`, `PartitionAware`
* Convert `IdentifiedDataSerializableFactory` and `PortableFactory` into functions
* Rename `input`/`output` arguments into `reader`/`writer` in Portable samples
* Convert enum values to strings for `ItemEventType`,  `OverflowPolicy`, `IterationType`
* Extract interface for `PagingPredicate`
* Fix `hazelcast.logging.level` property handling
* Remove configuration sample JSON files
* Move JSON schema into `./test`
* Also contains minor code style and jsdoc improvements

Corresponding PRD: https://hazelcast.atlassian.net/wiki/spaces/PM/pages/2016673900/Node.js+Client+Research+-+Redesign+Client+API

### CODE REVIEW

File: code_samples/org-website/EntryProcessorSample.js [hunk 1]
```diff
@@ -19,25 +19,18 @@ const { Client } = require('hazelcast-client');
...
-        return 1;
-    }
-
-    getClassId() {
-        return 9;
-    }
 }
 
-class EntryProcessorDataSerializableFactory() {
+class EntryProcessorDataSerializableFactory {
```
[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply code_samples/org-website/EntryProcessorSample.js]: Good catch. Fixed in ef0df9d2760547836a1a79e79ad337291b67cfd7

File: src/proxy/flakeid/AutoBatcher.ts [hunk 2]
```diff
@@ -28,6 +28,7 @@ export class Batch {
 
     constructor(validityMillis: number, base: Long, increment: Long, batchSize: number) {
         this.nextIdLong = base;
+        this.batchSize = batchSize;
```
[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply src/proxy/flakeid/AutoBatcher.ts]: Removed in ef0df9d2760547836a1a79e79ad337291b67cfd7

File: src/serialization/DefaultPredicates.ts [hunk 3]
```diff
@@ -454,7 +456,8 @@ export class PagingPredicate extends AbstractPredicate {
         } else if (page === anchorCount) {
             this.anchorList.push(anchorEntry);
         } else {
-            throw new RangeError('Anchor index is not correct, expected: ' + page + 'found: ' + anchorCount);
+            throw new RangeError('Anchor index is not correct, expected: '
```
[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply src/serialization/DefaultPredicates.ts]: Removed the method in ef0df9d2760547836a1a79e79ad337291b67cfd7

**Comment by puzpuzpuz** [comment]
@mdumandag I've just noticed that `PagingPredicate` interface included some methods meant to be private and fixed that. Could you check c22f40d23bdc02330ec3737a8d13a12be948c7f6?

**Comment by mdumandag** [comment]
It looks good, I also checked the Java side and saw that there is an extra `reset` method in their interface to reuse the predicate for other queries. Do you think that we should also add that?

**Comment by puzpuzpuz** [comment]
> It looks good, I also checked the Java side and saw that there is an extra `reset` method in their interface to reuse the predicate for other queries. Do you think that we should also add that?

I saw that one as well, but decided to leave it as is for now. I don't see a lot of value in predicate reuse in case of Node.js client. But if someone will ask for it, it's quick to port it. WDYT?

**Comment by mdumandag** [comment]
I am also fine with that

**Comment by puzpuzpuz** [comment]
@mdumandag thanks for the review. Going to merge this one now.

**Pull request merged**